### ISSUE TITLE
Directives such as `toctree` render incorrectly due to an extra newline during conversion

### ISSUE THREAD

**Issue by ethankaminski** [miyakogi/m2r#35]
I've run into an issue with Sphinx directives (`toctree` specifically) mixed into Markdown files. The directive itself works correctly, but `maxdepth` doesn't work. This appears to be because of an empty line between the directive itself and the `maxdepth` line, after conversion to `.rst`.

Looking at the m2r source (and testing a patch on my local copy), it looks like it's specifically a trailing newline in a `directive` method ([line 510 of m2r.py](https://github.com/miyakogi/m2r/blob/661a82ccfee48342dcbc3dcca4fd77e6a0a1a900/m2r.py#L510)). The fix looks simple, so I'll open a pull request for that.

For reproduction of the issue, take the following `.md` file:
```
# Header goes here

.. toctree::
   :maxdepth: 2
   :caption: In This Section:

   deploy_local_dev.md
   deploy_rhel7_centos7.md
   deploy_ubuntu.md
```

This converts to the following .rst source (tested with `m2r --dry-run`):
```
Header goes here
================

.. toctree::

   :maxdepth: 2
   :caption: In This Section:

   deploy_local_dev.md
   deploy_rhel7_centos7.md
   deploy_ubuntu.md
```

The extra newline between the `toctree` directive and the `:maxdepth: 2` line results in maxdepth being ignored, and also results in a warning being issued by Sphinx (`WARNING: toctree contains reference to nonexisting document ':maxdepth: 2'`).

**Comment by miyakogi** [comment]
Thank you for reporting issue and sending PR.
Fixed by #36.

**Comment by ethankaminski** [comment]
@miyakogi would a patch release of the `dev` branch be doable sometime soon, perchance?

We've got an open issue on our end related to this issue, and having it fixed in a numbered release would be better for us than the alternatives. (thanks for merging, as well!)
